src/audio/kpb.c

```diff
@@ -308,9 +312,9 @@ static int kpb_bind(struct comp_dev *dev, void *data)
 
 		if (sink_buf_id == buf_id) {
 			if (sink_buf_id == 0)
-				kpb->sel_sink = sink;
+				kpb->sinks[0] = sink;
 			else
-				kpb->host_sink = sink;
+				kpb->sinks[bu->extension.r.src_queue] = sink;
```

Comment:
should we check `bu->extension.r.src_queue` to avoid bad array access?

Reply:
ack, if this comes from host we need to validate.